openhexa/toolbox/hexa/api.py

```diff
@@ -0,0 +1,64 @@
+from requests import Session
+import typing
+
+
+class NotFound(Exception):
+    """Errors related to an element not found."""
+
+    pass
+
+
+class OpenHEXAClient:
+    def __init__(self, base_url):
+        self.url = base_url.rstrip("/")
+        self.session = Session()
+        self.session.headers.update({"Content-Type": "application/json", "User-Agent": "OpenHEXA Python Client"})
+
+    def authenticate(
+        self,
+        with_credentials: typing.Optional[tuple[str, str]] = None,
+        with_token: typing.Optional[str] = None,
+    ):
+        """
+        with_credentials: tuple of email and password
+        with_token: JWT token
+        """
+        if with_credentials:
+            resp = self._graphql_request(
+                """
+                mutation Login($input: LoginInput!) {
+                    login(input: $input) {
+                        success
+                    }
+                }
+            """,
+                {
+                    "input": {
+                        "email": with_credentials[0],
+                        "password": with_credentials[1],
+                    }
+                },
+            )
+            resp.raise_for_status()
+            data = resp.json()["data"]
+            if data["login"]["success"]:
```

Comment:
The correct way to ensure that the person has no 2FA is to check the `errors` on the result of the login mutation.